docs/reference/search/term-enum.asciidoc

```diff
@@ -0,0 +1,101 @@
+[[search-term-enum]]
+=== Term enum API
+
+The term enum API can be used to discover terms in the index that match
+a partial string. This is used for features like auto-complete:
+
+[source,console]
+--------------------------------------------------
+POST stackoverflow/_terms
+{
+    "field" : "tags",
+    "string" : "kiba"
+}
+--------------------------------------------------
+// TEST[setup:stackoverflow]
+
+
+The API returns the following response:
+
+[source,console-result]
+--------------------------------------------------
+{
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "failed": 0
+  },
+  "terms": [
+    "kibana"
+  ],
+  "timed_out" : false
```

Comment:
We also need to know if the result is complete or not ? For small cardinality field it can be useful to know that the response contains all the possible values for that input.

Reply:
"complete = true" will have some caveats e.g. if we skip indices on cold nodes